sdk/core/Azure.Core/src/Shared/EventSourceEvent.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+
+#nullable enable
+
+namespace Azure.Core.Shared
+{
+    internal readonly struct EventSourceEvent : IReadOnlyList<KeyValuePair<string, object?>>
```

Comment:
It might be helpful to add a comment describing what this type is and how it's used, if it's not too much trouble to add it.

Reply:
sure, done